controllers/openstackcluster_controller.go

```diff
@@ -627,3 +617,77 @@ func handleUpdateOSCError(openstackCluster *infrav1.OpenStackCluster, message er
 	openstackCluster.Status.FailureReason = &err
 	openstackCluster.Status.FailureMessage = pointer.String(message.Error())
 }
+
+// validateSubnets validates if the amount of IPv4 and IPv6 subnets is allowed.
+func validateSubnets(subnets []infrav1.Subnet) error {
```

Comment:
please add unit tests. I wrote them for you:

```golang
func Test_validateSubnets(t *testing.T) {
	tests := []struct {
		name    string
		subnets []infrav1.Subnet
		wantErr bool
	}{
		{
			name: "valid IPv4 and IPv6 subnets",
			subnets: []infrav1.Subnet{
				{
					CIDR: "192.168.0.0/24",
				},
				{
					CIDR: "2001:db8::/32",
				},
			},
			wantErr: false,
		},
		{
			name: "multiple IPv4 subnets",
			subnets: []infrav1.Subnet{
				{
					CIDR: "192.168.0.0/24",
				},
				{
					CIDR: "10.0.0.0/24",
				},
			},
			wantErr: true,
		},
		{
			name: "multiple IPv6 subnets",
			subnets: []infrav1.Subnet{
				{
					CIDR: "2001:db8::/32",
				},
				{
					CIDR: "2001:db8:1::/32",
				},
			},
			wantErr: true,
		},
		{
			name: "invalid IP address",
			subnets: []infrav1.Subnet{
				{
					CIDR: "192.168.0.0/24",
				},
				{
					CIDR: "2001:db8::/32",
				},
				{
					CIDR: "invalid",
				},
			},
			wantErr: true,
		},
	}

	for _, tt := range tests {
		t.Run(tt.name, func(t *testing.T) {
			err := validateSubnets(tt.subnets)
			if (err != nil) != tt.wantErr {
				t.Errorf("validateSubnets() error = %v, wantErr %v", err, tt.wantErr)
			}
		})
	}
}
```

Reply:
I update it a bit and included